### DISCUSSION TITLE
How can viewers not find html files or download 3d files?

### DISCUSSION THREAD

**Discussion by hmduc01** [google/model-viewer#2924] [Q&A]
Sorry for my English is not good
I'm worried about the security situation at my website ( [https://vietpoly.com/portfolio/1/](url) ) when the viewer only need to download the html file they can see the full content and download link of my .glb file, so how can i prevent that?

![Screenshot_9](https://user-images.githubusercontent.com/89718203/138805856-1976b1d5-2502-4f7b-ac5e-2e127ff93b47.png)

![Screenshot_10](https://user-images.githubusercontent.com/89718203/138805865-47d7020d-8cc9-45cb-8614-90e67abefcb7.png)

I know there is a way to do it because I checked at this website " [https://modelviewer.dev/](url) " and didnt see the link to download the .glb
So can anyone help me?. Many thanks.

**Comment by elalish** [comment]
Actually it's the same on modelviewer.dev, and even  when it's not obvious, you can always open the network tab of your browser's dev tools and you'll see all files that were fetched and can simply download any that are `*.glb`. I'm afraid the web just doesn't give you security here; a glTF is exactly like a JPEG; if you can see it, you can save it. So you need to think of security from the  standpoint of photographers; if you're selling your glTF, then only display a lower-res version on your website, or watermark it, or chase abusers down with the DMCA. There is no hardware decryption pipeline for 3D models like there is for video.
